lib/resty/healthcheck.lua

```diff
@@ -919,8 +960,32 @@ function checker:run_single_check(ip, port, hostname, hostheader)
   end
 
   local req_headers = self.checks.active.headers
-  local headers = table.concat(req_headers, "\r\n")
-  if #headers > 0 then
+  local headers
+  local headers_length = nkeys(req_headers)
+  if headers_length > 0 then
+    if is_array(req_headers) then
+      self:log(WARN, "array headers is deprecated")
+      headers = table.concat(req_headers, "\r\n")
+    else
+      headers = new_tab(0, headers_length)
+      local idx = 0
+      for k, v in pairs(req_headers) do
+          -- `k` is header name and `v` is header value
+          if type(v) == "table" then
+            for _, item in ipairs(v) do
+              idx = idx + 1
+              headers[idx] = escape_uri(k) .. ": " .. escape_uri(item)     
```

Comment:
some trailing whitespace here:

```suggestion
              headers[idx] = escape_uri(k) .. ": " .. escape_uri(item)
```